src/azuqua.js

```diff
@@ -0,0 +1,847 @@
+    _path = path.resolve(path.dirname(module.parent.filename), _path);
+    this.account = require(_path);
+  };
+
+  /**
+   * Async verion of loadConfig
+   * @example
+   * // Create 'empty' azuqua instance
+   * var azuqua = new Azuqua();
+   * // Load config from location
+   * azuqua.loadConfigAsync('./path/to/config.js', function(error, config) {
+   *   // Do something with results here
+   * });
+   * // azuqua instance now property configured
+   * @param {string} _path - The path of the config file
+   * @param {function} cb - Callback to be invoked when config is loaded. Callback is invoked with
+   * error, and config as arguments. 
+   */
+  loadConfigAsync (_path, cb) {
+    _path = path.resolve(path.dirname(module.parent.filename), _path);
+    fs.readFile(_path, (error, data) => {
+      if (error) {
+        cb(error, null);
+      } else {
+        try {
+          let config = JSON.parse(data);
+          this.account = config;
+          cb(null, config);
+        } catch (e) {
+          cb(e, null);
+        }
+      }
+    });
+  }
+
+  /**
+   * A generic function that generates the headers for a particular request
+   * @example
+   * // Produce an object with valid API headers generated from params (Notice it's static')
+   * Azuqua.generateHeaders('GET', 'flo/:alias/invoke', 'myaccesskey', 'myaccesssecret', {
+   *  data : {
+   *    name : 'Azuqua'
+   *  }
+   * })
+   * @param {string} method - The HTTP method
+   * @param {string} path - The path of the desired resource.
+   * @param {string} accessKey - The requesters access key
+   * @param {string} accessSecret - The requesters access secret
+   * @param {object} [data] - Additional data needed for the hash (if applicable)
+   */
+  static generateHeaders(method, path, accessKey, accessSecret, data) {
+    // Default params don't override null and we can't send null data
+    if (!data) {
+      data = '';
+    }
+
+    if (typeof data === 'object') {
+      data = JSON.stringify(data);
+    }
+
+    let pathQueryString;
+    if (method === 'GET' || method === 'DELETE') {
+      pathQueryString = data || '';
+    } else {
+      pathQueryString = data || '{}';
+    }
+
+
+    try {
+      let pathQueryStringHolder = JSON.parse(pathQueryString);
+      // The query string has all values in string form when it hits the server
+      // So id=833 would need to be id='833' when calculating the hash
+      pathQueryStringHolder = _.mapValues(pathQueryStringHolder, (value) => {
+        return value.toString();
+      })
+      pathQueryString = JSON.stringify(pathQueryStringHolder);
+    } catch(e) { 
+      let errProxy = new Error('Error mapping query string to string values');
+      return errProxy;
+    }
+
+
+    let timestamp = new Date().toISOString();
+
+    let meta = [method.toLowerCase(), path, timestamp].join(':') + pathQueryString;
+    let hash = crypto.createHmac('sha256', accessSecret).update(new Buffer(meta, 'utf-8')).digest('hex');
```

Comment:
`new Buffer()` is deprecated 

https://nodejs.org/api/buffer.html#buffer_new_buffer_array